### DISCUSSION TITLE
bash: get layout > create toggle function

### DISCUSSION THREAD

**Discussion by molleweide** [asmvik/yabai#1050] [Q&A] [answered]
Hi there,

I am quite new to yabai and I am wondering if it is possible to get the yabai layout in a bash script and eg. check if layout is stack and then toggle
it to bsp?

The reason why I am asking is because I am using kmonad for mapping my keyboard, and if I map a key to a yabai command
it outputs `yabai-msg: failed to connect to socket..` and so therefor my work around would then be to create a bash script and map it to my key and then toggle layout from this script.

**Comment by es183923** [comment, marked as answer]
A simple one-liner will suffice. 
For setting the space layout:
```sh
[ "$(yabai -m query --spaces --space | jq -r '.type')" = "stack" ] && yabai -m space --layout bsp
```
For the global layout:
```sh
[ "$(yabai -m config layout)" = "stack" ] && yabai -m config layout bsp
```

**Reply by molleweide** [reply]
woaw thank you very much sir!!

May I ask you one more thing: I created a test script that I call from kmonad and it also generated the same socket err described above. I feel pretty confident this has something to do with kmonad BUT is there any possibility this is an issue on yabai's end you think?

**Reply by es183923** [reply]
the error `yabai-msg: failed to connect to socket..` probably means that yabai isn't running

**Reply by es183923** [reply]
Does `pgrep yabai; echo $?` return a number, then zero?
eg:
```
$ pgrep yabai; echo $?
68768
0
```

**Reply by molleweide** [reply]
yes I can confirm it is running. I added pgrep to the scripts and now each key returns this to stdout:
```shell
╰ kmonad
Password:
connected
5802
0
yabai-msg: failed to connect to socket..
5802
0
yabai-msg: failed to connect to socket..
```

**Reply by es183923** [reply]
Do you have a config file in `~/.config/yabai/yabairc`?

**Reply by molleweide** [reply]
Yes! Yabai is working but just not via kmonad keys.

**Reply by es183923** [reply]
I'm out of ideas. I have no clue why yabai doesn't work with kmonad.

**Reply by molleweide** [reply]
No worries. I have posted an issue under kmonad. This probably has to do with how kmonad is execute shell. In kmonad there is special key for executing shell cmd on button press and I don't know how it is working under the hood.
I'll report back!!

**Reply by molleweide** [reply]
it turned out that kmonad was running commands as root user and therefore it didn't work fyi.
